src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -428,24 +436,46 @@ internal Task AbortAsync()
 
         private void KeepAliveTick()
         {
-            var timestamp = Stopwatch.GetTimestamp();
+            var currentTime = DateTime.UtcNow.Ticks;
             // Implements the keep-alive tick behavior
             // Each tick, we check if the time since the last send is larger than the keep alive duration (in ticks).
             // If it is, we send a ping frame, if not, we no-op on this tick. This means that in the worst case, the
             // true "ping rate" of the server could be (_hubOptions.KeepAliveInterval + HubEndPoint.KeepAliveTimerInterval),
             // because if the interval elapses right after the last tick of this timer, it won't be detected until the next tick.
-            if (timestamp - Interlocked.Read(ref _lastSendTimestamp) > _keepAliveDuration)
+            if (currentTime - Interlocked.Read(ref _lastSendTimestamp) > _keepAliveInterval)
             {
                 // Haven't sent a message for the entire keep-alive duration, so send a ping.
                 // If the transport channel is full, this will fail, but that's OK because
                 // adding a Ping message when the transport is full is unnecessary since the
                 // transport is still in the process of sending frames.
                 _ = TryWritePingAsync();
+            }
+        }
+
+        public void StartPeriodicallyCheckingForClientTimeout()
```

Comment:
Also, make this `internal`. This class is `public` so that certain user components can invoke methods on it, but I don't think we want them messing with Ping frames like this.

Reply:
For the name, I was trying to imply that the *checking* for potential timeouts was starting, rather than the timeout process itself.